lib/common/xml.c

```diff
@@ -751,6 +751,9 @@ free_xml_with_position(xmlNode *node, int position)
          */
         pcmk__xml_free_doc(doc);
         return pcmk_rc_ok;
+
+    } else if (doc == NULL) {
```

Comment:
pretty sure there are a lot of places we assume the doc is non-NULL, but of course clang picks a random one to complain about